### DISCUSSION TITLE
Does CoreWCF implement behaviour extensions?

### DISCUSSION THREAD

**Discussion by DublinDel** [CoreWCF/CoreWCF#307] [Q&A]
I have message inspectors which use custom behaviour extensions to create customer behaviours on endpoints using the IEndpointBehavior interface. Is this planned for CoreWCF or is there a different way of adding behaviours into Endpoints

[BehaviourExtensionExample.txt](https://github.com/CoreWCF/CoreWCF/files/5970744/BehaviourExtensionExample.txt)

**Comment by mconnew** [comment]
The easiest way is to use an extension method on IServiceBuilder called ConfigureServiceHostBase which takes a delegate. So you can do something like this:
```c#
            public void Configure(IApplicationBuilder app)
            {
                app.UseServiceModel(builder =>
                {
                    builder.AddService<Services.EchoService>();
                    builder.AddServiceEndpoint<Services.EchoService, ServiceContract.IEchoService>(new CoreWCF.BasicHttpBinding(), "/BasicWcfService/basichttp.svc");
                    builder.ConfigureServiceHostBase<Services.EchoService>(serviceHost =>
                    {
                        IEndpointBehavior behavior = new MyCustomSessionBehavior()
                        foreach(var endpoint in serviceHostBase.Description.Endpoints)
                        {
                            endpoint.EndpointBehaviors.Add(behavior);
                        }
                    });
                });
            }
```

Currently this won't work until PR #308 is merged as the provided delegate was being executed too late and that PR fixes this.

**Reply by mconnew** [reply]
Package version 0.1.0-preview-2 is now on nuget with this fixed.

**Reply by DublinDel** [reply]
Very much obliged Matt
